crates/bevy_ecs/src/system/query.rs

```diff
@@ -226,3 +260,23 @@ pub enum QueryComponentError {
     #[error("The requested entity does not exist.")]
     NoSuchEntity,
 }
+
+#[derive(Error)]
+pub enum QuerySingleError<'a, Q: WorldQuery> {
+    #[error("No entities fit the query {0}")]
+    NoEntities(&'static str),
```

Comment:
Would it make sense to have the same layout for `NoEntities` as for `MultipleEntities`?
This removes the slightly strange asymmetry and would allow to extend the error information in the future.